src/lib.rs

```diff
@@ -126,14 +126,14 @@ where
     let mut buf = Vec::with_capacity(amt);
```

Comment:
Could you replace this line with the following code.


```rust
let mut buf = vec![0; amt];
```

The reason is that the intent of the code is more readable, and the compiler optimizations may translate it into a more efficient allocation function call (Rust's allocator interface has `GlobalAlloc::alloc_zeroed()`).